ios/BrightcovePlayer.m

```diff
@@ -129,20 +143,110 @@ - (void)playbackController:(id<BCOVPlaybackController>)controller playbackSessio
         if (self.onReady) {
             self.onReady(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackReady")
+                                 });
+        }
     } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPlay) {
         _playing = true;
         if (self.onPlay) {
             self.onPlay(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackPlay")
+                                 });
+        }
     } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPause) {
         _playing = false;
         if (self.onPause) {
             self.onPause(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackPause")
+                                 });
+        }
     } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventEnd) {
         if (self.onEnd) {
             self.onEnd(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackEnd")
+                                 });
+        }
+    } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventFail) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("loadFail"),
+                                 @"error": lifecycleEvent.properties[kBCOVPlaybackSessionLifecycleEventError]
```

Comment:
From the documentation, this seems to return the NSError object. Is that true? If so, how does that translate? I think we probably need to be consistent and see how other APIs transmit the NSError, or at least make the Error have a error code and string for the error. Also, I think this can be "nil" and you can't put a nil object into the Dictionary. From the documentation it seems as though it is optional. So we should probably check for that before putting it into the dictionary. If nil, then use a generic error.

Reply:
I'm not sure what it returns.  I'll look at the Brightcove SDK and see what it's generating.  From what I read in the docs, it looks like an 'error' property, but an NSError makes sense.  I'll see what I can learn.

Reply:
The calls above (findVideoWithVideoID and findVideoWithReferenceID) return NSError and I added them to the the messages we are returning.
These errors (returning: BCOVPlaybackSessionLifecycleEvent) include the error as a property in  'properties' dictionary.  I haven't found a way to trigger these generic errors to see what actually comes back.
I only returned the error on the events that said they return an error property.  I'll check for nil just in case.